src/sjcl-custom/sjcl-ecdsa-signdeterministic.js

```diff
@@ -0,0 +1,119 @@
+'use strict';
+let sjcl = require('sjcl');
+/* eslint-disable new-cap */
+
+sjcl.ecc.ecdsa.secretKey.prototype.generateK =
+function(hash, hashObject) {
+
+  let curve = this._curve;
+  let qlen = sjcl.bitArray.bitLength(curve.r.toBits());
+
+  /* Utility functions */
+  /* used to generate k and v */
+  function repeat(str, times) {
+    return (new Array(times + 1)).join(str);
+  }
+  function bits2int(bits) {
+
+    let blen = sjcl.bitArray.bitLength(bits);
+
+    if (blen > qlen) {
+      return sjcl.bn.fromBits(sjcl.bitArray.bitSlice(bits, 0, qlen));
+    } else if (qlen < blen) {
+      return sjcl.bn.fromBits(sjcl.bitArray.concat(
+        sjcl.codec.hex.toBits(repeat('0', Math.ceil((qlen - blen) / 4))),
```

Comment:
You may notice that it's hard to directly test these units cause they aren't exposed and get state from a closure.